ckan/logic/action/update.py

```diff
@@ -777,10 +783,151 @@ def _group_or_org_update(context, data_dict, is_org=False):
+            _get_action(action)(reindex_context, {'id': group.id})
 
     return model_dictize.group_dictize(group, context)
 
 
+# (canada fork only): background search index rebuilding
+#TODO: upstream contrib!!
+def _group_or_org_reindex(context, data_dict, is_org=False):
+    model = context['model']
+    session = context['session']
+    id = _get_or_bust(data_dict, 'id')
+
+    group = model.Group.get(id)
+    context["group"] = group
+    if group is None:
+        raise NotFound('Group was not found.')
+
+    data_dict['type'] = group.type
+
+    if is_org:
+        _check_access('reindex_organization_datasets', context, data_dict)
+    else:
+        _check_access('reindex_group_datasets', context, data_dict)
+
+    results = session.query(model.Package.id)\
+        .filter(model.Package.owner_org == group.id)\
+        .filter(model.Package.state == 'active').all()
+
+    if not results:
+        return
+
+    task = {
+        'entity_id': group.id,
+        'entity_type': 'group',
+        'task_type': 'search_rebuild',
+        'last_updated': str(datetime.datetime.now(datetime.timezone.utc)),
+        'state': 'submitting',
+        'key': 'search_rebuild',
+        'value': '{}',
+        'error': '{}',
+    }
+    try:
+        existing_task = _get_action('task_status_show')(context, {'entity_id': group.id,
+                                                                  'task_type': 'search_rebuild',
+                                                                  'key': 'search_rebuild'})
+        if existing_task.get('state') == 'pending':
+            log.info('A pending task (%s) was found for this %s (%s). Skipping this duplicate task...',
+                     existing_task['id'], 'Organization' if is_org else 'Group', group.name)
+            return False
+        task['id'] = existing_task['id']
+    except NotFound:
+        pass
+
+    _get_action('task_status_update')({'session': model.meta.create_local_session(), 'ignore_auth': True}, task)
+
+    dataset_ids = [dataset.id for dataset in results]
+    job_title = 'Rebuild Dataset Indecies for %s: %s' % ('Organization' if is_org else 'Group', group.name)
+    queue_name = plugins.toolkit.config.get('ckan.search.rebuild_queue_name', 'search_rebuild')
+
+    job = plugins.toolkit.enqueue_job(fn=search.rebuild, title=job_title, queue=queue_name,
+                                      kwargs={'force': True, 'package_ids': dataset_ids, 'in_background': True, 'group_id': group.id})
+
+    task['value'] =  json.dumps({'job_id': job.id, 'total': len(dataset_ids), 'indexed': 0})
+    task['state'] = 'pending'
+    task['last_updated'] = str(datetime.datetime.now(datetime.timezone.utc))
+
+    _get_action('task_status_update')({'session': model.meta.create_local_session(), 'ignore_auth': True}, task)
+
+    return dictize_job(job)
+
+
+# (canada fork only): background search index rebuilding
+#TODO: upstream contrib!!
+def reindex_organization_datasets(context, data_dict):
```

Comment:
these are creating background jobs so maybe the naming and help text should indicate that?